### PR TITLE
RDK-38125 : Added Unit tests for FrameBuffer based ScreenCapture.

### PR DESCRIPTION
**Pull request by sborushevsky**

### CODE REVIEW

**Comment by npoltorapavlo** [comment]
please fix GMOCK WARNING-s produced by this pr
see https://github.com/rdkcentral/rdkservices/issues/3596

File: Tests/CMakeLists.txt [hunk 1]
```diff
@@ -104,7 +106,9 @@ target_link_libraries(${PROJECT_NAME}
         ${NAMESPACE}DeviceIdentification
         ${NAMESPACE}FrameRate
         ${NAMESPACE}AVInput
+        ${NAMESPACE}ScreenCapture
         ${NAMESPACE}Telemetry
+        ${NAMESPACE}ScreenCapture
```
Comment (npoltorapavlo) [inline Tests/CMakeLists.txt:111]: duplicate

File: Tests/tests/test_ScreenCapture.cpp [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](FBContext* fbctx, VncServerFramebufferAPI* server, void* serverctx) {
+                if (nullptr != server)
+                {
+                    server->framebufferUpdateReady(&fbcontext);
+                    server->framebufferDetailsChanged(&fbcontext, nullptr, 1280, 720, 5120, nullptr);
+                    server->paletteChanged(&fbcontext, nullptr);
+                    server->logMsg(&fbcontext, "msg");
+                }
```
Comment (npoltorapavlo) [inline Tests/tests/test_ScreenCapture.cpp:103]: as a reader, I don't understand what these lines are for, even after checking the code. is it necessary?

File: Tests/tests/test_ScreenCapture.cpp [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+    ON_CALL(frameBufferApiImplMock, fbGetStride(::testing::_))
+        .WillByDefault(
+            ::testing::Return(5120));
+
+    EXPECT_CALL(frameBufferApiImplMock, fbGetFramebuffer(::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](FBContext* fbctx) {
+                return frameBuffer;
```
Comment (npoltorapavlo) [inline Tests/tests/test_ScreenCapture.cpp:127]: please use `Return(frameBuffer)` and `using testing::Return;` where it simply  returns

**Review by npoltorapavlo — CHANGES_REQUESTED** [review]

File: Tests/tests/test_ScreenCapture.cpp (hunk 2 above)
Comment (npoltorapavlo) [reply Tests/tests/test_ScreenCapture.cpp:105]: if these lines don't verify anything please remove them to improve readability

Comment (sborushevsky) [reply Tests/tests/test_ScreenCapture.cpp]: Removed.

**Review by npoltorapavlo — APPROVED** [review]

**Pull request merged**